src/Files.App/UserControls/OpacityIcon.xaml.cs

```diff
@@ -11,15 +11,28 @@ public OpacityIcon()
 			InitializeComponent();
 		}
 
+
+		public bool IsSelected
+		{
+			get => (bool)GetValue(IsSelectedProperty);
+			set => SetValue(IsSelectedProperty, value);
+		}
+
+		public static readonly DependencyProperty IsSelectedProperty =
+			DependencyProperty.Register(nameof(IsSelected), typeof(bool), typeof(OpacityIcon), new PropertyMetadata(null));
+
 		private void IsEnabledChange(DependencyObject sender, DependencyProperty dp)
 		{
 			string state = sender.GetValue(dp) is false ? "Disabled" : "Normal";
-			VisualStateManager.GoToState(this, state, true);
+
+			if (state == "Normal" && IsSelected)
+				VisualStateManager.GoToState(this, "Selected", true);
+			else
+				VisualStateManager.GoToState(this, state, true);
```

Comment:
I think something like this would be better to read:
```c#
if(IsSelected) 
    VisualStateManager.GoToState(this, "Selected", true);
if(sender.GetValue(IsEnabledProperty))
    VisualStateManager.GoToState(this, "Normal", true);
VisualStateManager.GoToState(this, "Disabled", true)
```

Reply:
> I assume there is no use case for selected and simultaneously disabled right?

That's right, but it can be selected and enabled at the same time in which `Selected` should have priority.

Reply:
@chingucoding have you tested if that works? I didn't test but the logic doesn't look right.

Reply:
I did not and it also wouldn't. I think the following would work as expected:
```c#
			if(sender.GetValue(dp) is false)
				VisualStateManager.GoToState(this, "Disabled", true);
			else if (IsSelected)
				VisualStateManager.GoToState(this, "Selected", true);
			else
				VisualStateManager.GoToState(this, "Normal", true);
```

Reply:
This looks good, thanks!